gthockey-iOS/gthockey-iOS/CustomUIComponents/PillButton.swift

```diff
@@ -57,4 +58,36 @@ class PillButton: UIButton {
         fatalError("init(coder:) has not been implemented")
     }
 
+    // MARK: Loading Animation
+
+    private var originalButtonText: String?
+
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        activityIndicator.color = .black
+        addSubview(activityIndicator)
+
+        NSLayoutConstraint.activate([
```

Comment:
Take the constraints and move it into its own function which should be called `override func updateConstraints()`. Make sure to call it after `addSubview()`.

*This is similar to the flow in classes like `SignupView.swift` and `LoginView.swift`*